arch/risc-v/src/bl602/bl602_netdev.c

```diff
@@ -0,0 +1,2113 @@
+ *   Handle network IOCTL commands directed to this device.
+ *
+ * Input Parameters:
+ *   dev - Reference to the NuttX driver state structure
+ *   cmd - The IOCTL command
+ *   arg - The argument for the IOCTL command
+ *
+ * Returned Value:
+ *   OK on success; Negated errno on failure.
+ *
+ * Assumptions:
+ *   The network is locked.
+ *
+ ****************************************************************************/
+
+static int
+bl602_net_ioctl(FAR struct net_driver_s *dev, int cmd, unsigned long arg)
+{
+  FAR struct bl602_net_driver_s *priv =
+    (FAR struct bl602_net_driver_s *)dev->d_private;
+  int ret = -ENOSYS;
+
+  /* Decode and dispatch the driver-specific IOCTL command */
+
+  switch (cmd)
+    {
+    case SIOCSIWSCAN:
+      do
+        {
+          struct iwreq *             req  = (struct iwreq *)arg;
+          struct scan_parse_param_s *para = NULL;
+
+          para = kmm_malloc(sizeof(struct scan_parse_param_s));
+          if (para == NULL)
+            {
+              return -ENOMEM;
+            }
+
+          para->priv = priv;
+          if (req->u.data.flags)
+            {
+              para->flags = req->u.data.flags;
+
+              assert(req->u.data.pointer != NULL);
+              para->scan_req = *(struct iw_scan_req *)req->u.data.pointer;
+            }
+          else
+            {
+              para->flags = 0;
+            }
+
+          if (sem_trywait(&g_wifi_scan_sem) == 0)
+            {
+              wifi_mgmr_scan(para, scan_complete_indicate);
+              return OK;
+            }
+          else
+            {
+              return -EBUSY;
+            }
+        }
+      while (0);
+      break;
+
+    case SIOCGIWSCAN:
+      do
+        {
+          struct iwreq *req = (struct iwreq *)arg;
+
+          sem_wait(&g_wifi_scan_sem);
+
+          if (priv->scan_result_len == 0)
+            {
+              req->u.data.length = 0;
+              sem_post(&g_wifi_scan_sem);
+              return OK;
+            }
+
+          ret = format_scan_result_to_wapi(req, priv->scan_result_len);
+          sem_post(&g_wifi_scan_sem);
+          return ret;
+        }
+      while (0);
+      break;
+
+    case SIOCSIFHWADDR: /* Set device MAC address */
+      return -ENOSYS;
+      break;
+
+    case SIOCSIWAUTH:
+      /* FIXME not support set auth param now, return OK to support
+       * wapi reconnect command
+       */
+
+      return OK;
+      break;
+
+    case SIOCSIWENCODEEXT: /* Set psk */
+      do
+        {
+          struct iwreq *        req        = (struct iwreq *)arg;
+          struct iw_encode_ext *ext        = req->u.encoding.pointer;
+          char *                passphrase = kmm_malloc(ext->key_len + 1);
+          if (passphrase == NULL)
+            {
+              return -ENOMEM;
+            }
+
+          strncpy(passphrase, (char *)ext->key, ext->key_len);
+          passphrase[ext->key_len] = 0;
+          syslog(LOG_INFO, "passphrase: %s\r\n", passphrase);
```

Comment:
Should probably not log the passphrase. Also in the driver winfo instead of syslog